### DISCUSSION TITLE
v1.2.45 ICC Profile for WebP, JPEG XL, HEIF/HEIC, AVIF.  Bugfix: AVX2+4K resolution freeze

### DISCUSSION THREAD

**Discussion by sylikc** [sylikc/jpegview#160] [General]
## 1.2.45

This is primarily a bugfix release, with the new ICC profile support feature.

_(Minor version bumped to reflect the long awaited bugfix for JPEGView freezing when panning images on newer computers with AVX2 support and 4K+ resolutions.)_

### New features:
- Add support for **ICC Profiles embedded in WebP, JPEG XL, HEIF/HEIC and AVIF images**
  Set INI option `UseEmbeddedColorProfiles=true` to enable.
  _NOTE: Currently, this option will disable Animated PNG due to usage of GDI+ (see 1.1.44 release notes)_
  PR #156, thanks [qbnu](https://github.com/qbnu)

### Bugs fixed:
- Fixed: Freeze in JPEGView when `CPUType` set/auto-detected as `AVX2`, `HighQualityResampling=true`, and `window width > 3200 pixels` (either spanned over multi-monitor or on a single monitor).
  _(This outstanding bug had been extensively reported since v1.0.35+.  Thanks to everyone who provided samples, videos, and system configurations to help identify the specific conditions necessary to trigger this bug!)_
- Fixed: Regression introduced in 1.1.44 which caused `Fit Image to Screen` to create a tiny window instead of a window which filled the screen.
- Fixed: In certain cases with certain image, after zooming out to 0% (as far as allowed), zooming in did not work and got stuck at 0%
- Fixed: `FileNameFormat` INI option `%filesize%` substitution now shows MB when exceeding 1 MB, not 100 MB

### Other changes:
- Updated French translation, thanks [LaurentGrenet](https://github.com/LaurentGrenet)

## SHA-256 Checksums

Verify the release with these checksums on the release files:
```sha256sum
fde4558184a148e88bf491dad335067da7ad7d731d43bfe94d6319cac9c7d574 *JPEGView32_1.2.45.7z
8119296144fcd97d4be369293402bdf58389fbf8681de45d9d40a9619b8eb136 *JPEGView32_en-us_1.2.45.msi
aa3ace8474d1b16fa252d20c2fc41517412035b46a20d95f5e06dd2918a178b8 *JPEGView32_WinXP_1.2.45.zip
5ba2c528a88fc04f43c0daaa78996c2baf56869a846b9266796ff13bb7039613 *JPEGView64_1.2.45.7z
b1d496f660256a747c886c078cfd38424dfe6d0c79bb15f8e35d6a73092de039 *JPEGView64_en-us_1.2.45.msi
d18d2e04eb7c9919199f76a0a0c303f05dda810b55e1d678870a0de006a3f393 *JPEGView_1.2.45.7z
fef4bac6977c68e05b7e1f02b89ce92088a7b1d5157fac89764b8774bfe91a2d *JPEGView_1.2.45.zip
```

<hr /><em>This discussion was created from the release <a href='https://github.com/sylikc/jpegview/releases/tag/v1.2.45'>v1.2.45 ICC Profile for WebP, JPEG XL, HEIF/HEIC, AVIF.  Bugfix: AVX2+4K resolution freeze</a>.</em>

**Comment by LaurentGrenet** [comment]
Why 1.**_2_**.45 ? For a minor version, 1.**_1_**.45 would have been enough...

Nevertheless, no trouble with that ;-)

**Reply by sylikc** [reply]
the "minor" version bump which is the second number is to reflect the big bugfix.  There were 4 open issues on the AVX2+4K bugfix, with the earliest one opened almost 2 years ago.  I felt that was a pretty good reason to move the version up one.

**Comment by qbnu** [comment]
The MSI is missing lcms2.dll, I forgot to mention that.

**Reply by sylikc** [reply]
🙄 crap lol.  Thanks for keeping me in check qbnu!

I'll just replace the MSI files and not change the version lol

**Reply by sylikc** [reply]
Ok, I fixed it.  I re-released the MSI files without touching the version number, added a note, and updated all the checksums and files.  Thanks again...

Luckily I didn't send this into WinGet yet 😑

**Comment by nao9797** [comment]
You added HEIC Support!??! Very Awesome TYSM!!!
